src/hotspot/share/utilities/debug.cpp

```diff
@@ -238,6 +238,33 @@ void report_vm_error(const char* file, int line, const char* error_msg)
   report_vm_error(file, line, error_msg, "%s", "");
 }
 
+static char _detail_msg[1024];
+
+static void print_error_for_unit_test(const char* message, const char* detail_fmt, va_list detail_args) {
+#ifdef ASSERT
+  if (ExecutingUnitTests) {
+    if (detail_fmt != NULL) {
```

Comment:
You miss va_copy for detail_args here.

Reply:
va_list detail_args_copy;
      va_copy(detail_args_copy, detail_args);
      jio_vsnprintf(_detail_msg, sizeof(_detail_msg), detail_fmt, detail_args_copy);

I fixed this but I didn't see the message garbled (?)

You're right in your comment that we are testing the assert message through print_error_for_unit_test.  I think the original motivation was to move the test that checked for a specific message out of the error handling tests, and these other ones just came for the ride.

Reply:
Thats fine, I like this. Its a good simplification.